ui/src/hooks.ts

```diff
@@ -145,12 +150,45 @@ export function useConceptSet() {
   return useOptionalConceptSet(true) as NonNullable<tanagraUI.UIConceptSet>;
 }
 
+function useOptionalFeatureSet(throwOnUnknown: boolean) {
+  const featureSet = useContext(FeatureSetContext)?.state?.present;
+  if (throwOnUnknown && !featureSet) {
+    throw new PathError(`No valid feature set in current context.`);
+  }
+  return featureSet;
+}
+
+export function useFeatureSet() {
+  return useOptionalFeatureSet(true) as NonNullable<FeatureSet>;
+}
+
+function useOptionalFeatureSetAndCriteria(throwOnUnknown: boolean) {
```

Comment:
Is this for listing the possible data features in the dataset builder page?

Reply:
No, this is referencing the criteria and feature set it belongs to currently selected via the URL, for use while editing the criteria. The catch is that context values can always potentially be null, since there's no way for the compiler to validate how you compose your components. So this function returns those values optionally in case something needs to check if it's valid, but there's a wrapper that throws so the criteria editing page doesn't have no worry about null values when they're expected to be valid based on routing.